### PR TITLE
Sils/resas

### PR DESCRIPTION
**Pull request by sils**
### Check list

- [ ] I have rebased properly.
      [Please see this for a tutorial.](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing)
- [ ] I have gone through the
      [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html)
      and I've followed them:
 - [ ] I have followed the
       [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
 - [ ] I have followed the
       [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
 - [ ] I have included the issue URL with the  `Fixes` keyword if the
       commit fixes a *real bug* and the `Closes` keyword if it adds a
       feature/enhancement.
       [See more here.](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference)
- [ ] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html)
      and they all pass. If any CI (below) is not green, please fix them. If
      GitMate issues appear you will have to amend your commits, pushing new
      commits on top is not sufficient!

### CODE REVIEW

**Comment by Asnelchristian** [comment]
ack https://github.com/coala/coala/pull/2958/commits/20f49368aacb1aab340002e8a0a95c7cf9ffbd27

**Comment by Asnelchristian** [comment]
ack https://github.com/coala/coala/pull/2958/commits/48227138a1e0f245ff7ae746fad389a976eb9de0

**Comment by Asnelchristian** [comment]
ack https://github.com/coala/coala/pull/2958/commits/1d03d967cbd6513d2a1b20043d54c474df92c873

File: coalib/bearlib/aspects/__init__.py [hunk 1]
```diff
@@ -237,6 +237,12 @@ class Aspect:
     >>> Root.Redundancy.Clone.__qualname__
     'Redundancy.Clone'
 
+    As this should also identifie the aspect, it serves as string representation
```
Comment (adtac) [inline coalib/bearlib/aspects/__init__.py]: `identify`

File: coalib/results/Result.py [hunk 2]
```diff
@@ -61,6 +65,11 @@ def __init__(self,
         :param confidence:
             A number between 0 and 100 describing the likelihood of this result
             being a real issue.
+        :param aspect:
+            An Aspect object which this result is associated to. Note that this
+            should be a leaf of the aspect tree! (If you have a node, spend
```
Comment (adtac) [inline coalib/results/Result.py]: make a check like `assert aspects.subaspects == {}` to enforce this?

File: coalib/bearlib/aspects/Metadata.py [hunk 3]
```diff
@@ -27,6 +27,23 @@
...
+        Your commit message serves as important documentation for your source
+        code.
+        """,
+        example="(no text at all)",
+        example_language="English",
+        importance_reason="""
+        An empty commit message shows the lack of documentation for your
+        change.
+        """,
+        fix_suggestions="Write a commit message."
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: I actually laughed at this :D so blunt

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2958/commits/9981f8c8f1a3fba35ada799e0015ae219db72da2 https://github.com/coala/coala/pull/2958/commits/b4e88bd1b3c8b60b477968597d42a679c03abb51

File: coalib/bearlib/aspects/Metadata.py [hunk 4]
```diff
@@ -219,3 +219,23 @@
...
+        """,
+        example_language="English",
+        importance_reason="""
+        Having a nonempty commit body is important if you consistently want
+        elaborate documentation on all commits.
+        """,
+        fix_suggestions="""
+        Write a commit message with a body.
+        """
+    )
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: shouldn't there be a setting here that we can disable?

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2958/commits/ce48cd91f917b891c65416802dc3db0cd763a04a

File: coalib/bearlib/aspects/Metadata.py [hunk 5]
```diff
@@ -239,3 +239,30 @@
...
+        """,
+        example="Some people just write very long commit messages. Too long. "
+                "Way too much actually. If they would just break their lines!",
+        example_language="English",
+        importance_reason="""
+        Git and platforms like GitHub usually break everything beyond 72
+        characters, making a message containing longer lines hard to read.
+        """,
+        fix_suggestions="""
+        Simply break your lines right before you hit the 72 char border.
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: don't hardcode 72

File: coalib/bearlib/aspects/Metadata.py (hunk 4 above)
Comment (sils) [inline coalib/bearlib/aspects/Metadata.py]: mind that aspects are enabled/disabled! Settings only configure behaviour

File: coalib/bearlib/aspects/Metadata.py [hunk 6]
```diff
@@ -239,3 +239,30 @@
         """
     )
 )
+
+Root.Metadata.CommitMessage.Body.new_subaspect(
+    "Length",
+    AspectDocumentation(
+        definition="""
+        The length of every your commit message body lines.
```
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: remove `every`

also `commit message's`

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2958/commits/1134d08e4c5e40d5137a13b7f9c414295a1590b6 https://github.com/coala/coala/pull/2958/commits/48261d21f83e68b171394fd6b1de67d8e3ae5fc7

File: coalib/bearlib/aspects/Metadata.py (hunk 4 above)
Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: Description should be unbiased - `Describes whether the commit body exists or is empty` or something like that IMO

**Comment by sils** [comment]
reack 9c2f4a63a2717469211b11b9ee1e4742171f26e1 c47809980ee619685377354a446ab686554b3c48 f058a2cd255e7bdb07ed4a566bb3477e6e751961

File: coalib/bearlib/aspects/Metadata.py (hunk 4 above)
Comment (sils) [inline coalib/bearlib/aspects/Metadata.py]: this check is biased, a check for no commit message body doesn't make sense, does it?

Comment (adtac) [inline coalib/bearlib/aspects/Metadata.py]: hmm alright

but you never know :P

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/2958/commits/b9fa40806b57d94f8b52f75c1af9826dfafe6daa https://github.com/coala/coala/pull/2958/commits/185da2273709609b5d353c2c26646b0a6938064c https://github.com/coala/coala/pull/2958/commits/598ce35c1c8c150920a88c8026be7c2e301ed050 https://github.com/coala/coala/pull/2958/commits/84b762f524d44852dae6dbbe9c9b662f2852e7ba https://github.com/coala/coala/pull/2958/commits/57199262772f8a6f7d339cfb339043e30f1dbc93 https://github.com/coala/coala/pull/2958/commits/d27f568e7846d934871d27e4ceb243b211ce1efa

**Comment by sils** [comment]
@rultor merge

**Pull request merged**